plugins/pencilblue/templates/angular/admin/content/articles/article_form.html

```diff
@@ -35,6 +36,21 @@
 			});
 		};
 
+		$scope.generateUrl = function() {
+			$scope.article.url = $scope.article.headline.replace(/\s+/g, '-').toLowerCase();
+			if($scope.article.url){
+				$scope.urlGenerated = true;
+			} else {
```

Comment:
The generate button is only active if there's a valid headline, which means that a url will always be generated.